networkx/utils/backends.py

```diff
@@ -126,45 +126,97 @@
+``should_run`` are used by the dispatcher to determine whether to use the backend
+function. If the number of nodes is small, it might be faster to run the NetworkX
+version of the function. This is how backends can provide info about whether to run.
+
+Falling Back to NetworkX:
+
+If none of the backends are appropriate, we "fall back" to the NetworkX function.
+That means we resolve the backends of all input graphs and if all are NetworkX
+graphs we call the NetworkX function. If any are not NetworkX graphs, we raise
+an exception unless the `fallback_to_nx` config is set. If it is, we convert all
+graph types to NetworkX graph types before calling the NetworkX function.
+
+Functions that mutate the graph:
+
+Any function decorated with the option that indicates it mutates the graph goes through
+a slightly different path to automatically find backends. These functions typically
+generate a graph, or add attributes or change the graph structure. The config
+`backend_priority.generators` holds a list of backend names similar to the config
+`backend_priority`. The process is similar for finding a matching backend. Once found,
+the backend function is called and a backend graph is returned (instead of a NetworkX
+graph). You can then use this backend graph in any function supported by the backend.
+And you can use it for functions not supported by the backend if you set the config
+`fallback_to_nx` to allow it to convert the backend graph to a NetworkX graph before
+calling the function.
+
+Optional keyword arguments:
+
+Backends can add optional keyword parameters to NetworkX functions to allow you to
+control aspects of the backend algorithm. Thus the function signatures can be extended
+beyond the NetworkX function signature. For example, the ``parallel`` backend might
+have a parameter to specify how many CPUs to use. These parameters are collected
+by the dispatchable decorator code at the start of the function call and used when
+calling the backend function.
+
+Existing Backends:
+
+NetworkX does not know all the backends that have been created.
+In fact, the NetworkX library does not need to know that a backend exists
+for it to work. As long as the backend package creates the ``entry_point``, and
+provides the correct interface, it will be called when the user requests
+it using one of the three approaches described above. Some backends have
+been working with the NetworkX developers to ensure smooth operation.
+They are the following:
+
+- `graphblas <https://github.com/python-graphblas/graphblas-algorithms>`_:
+  OpenMP-enabled sparse linear algebra backend.
+- `cugraph <https://github.com/rapidsai/cugraph/tree/branch-24.04/python/nx-cugraph>`_:
+  GPU-accelerated backend.
+- `parallel <https://github.com/networkx/nx-parallel>`_:
+  Parallel backend for NetworkX algorithms.
+- `loopback <https://github.com/networkx/networkx/blob/main/pyproject.toml#L53>`_:
+  It's for testing purposes only and is not a real backend.
```

Comment:
I removed this in favor of just directing the reader to the backends "gallery" page. I figured the backends page would be more up-to-date anyway.  Did you want this list back though?